### PR TITLE
Allow `Char` to be compiled

### PR DESCRIPTION
**Pull request by henriquegemignani**
Also adds StaticPaddedString, which is used by BMSLD's FunctionArgument

### CODE REVIEW

**Comment by henriquegemignani** [comment]
Changed how this works by adding `StaticPaddedString`.

File: src/mercury_engine_data_structures/common_types.py [hunk 1]
```diff
@@ -127,6 +130,7 @@ def _emitparse(self, code):
 
         block = f"""
             def {fname}(io, this):
+                # {self.name} ({self})
```
Comment (Miepee) [inline src/mercury_engine_data_structures/common_types.py:133]: what is this for?

Comment (henriquegemignani) [reply src/mercury_engine_data_structures/common_types.py:133]: oh, just makes it slightly easier to debug the generated files

Comment (Miepee) [reply src/mercury_engine_data_structures/common_types.py:133]: ah ok

**Review by Miepee — APPROVED** [review]

**Pull request merged**